src/ring_theory/algebra.lean

```diff
@@ -650,6 +650,22 @@ def under {R : Type u} {A : Type v} [comm_ring R] [comm_ring A]
 lemma mul_mem (A' : subalgebra R A) (x y : A) :
   x ∈ A' → y ∈ A' → x * y ∈ A' := @is_submonoid.mul_mem A _ A' _ x y
 
+/-- Transport a subalgebra via an algebra homomorphism. -/
+def map (S : subalgebra R A) (f : A →ₐ[R] B) : subalgebra R B :=
+{ carrier := f '' (S : set A),
+  subring := ring_hom.is_subring_image (f : A →+* B) _,
+  range_le' := λ x ⟨r, hr⟩, ⟨algebra_map R A r, S.range_le ⟨r, rfl⟩, by rw [f.commutes, hr]⟩ }
+
+/-- Preimage of a subalgebra under an algebra homomorphism. -/
+def comap' (S : subalgebra R B) (f : A →ₐ[R] B) : subalgebra R A :=
```

Comment:
Is there already another `subalgebra.comap`? Or is this in the `algebra` namespace? I would think this belongs in `subalgebra.*`, right?

Reply:
```
subalgebra.comap :
  Π {R : Type u_1} {S : Type u_2} {A : Type u_3} [_inst_6 : comm_ring R] [_inst_7 : comm_ring S] [_inst_8 : ring A]
  [_inst_9 : algebra R S] [_inst_10 : algebra S A], subalgebra S A → subalgebra R (algebra.comap R S A)
```

Reply:
Thanks